ocaml/xapi/xapi_pool_update.ml

```diff
@@ -302,7 +302,25 @@ let verify update_info update_path =
     ) [update_xml_path; repomd_xml_path];
   debug "Verify signature OK for pool update uuid: %s by key: %s" update_info.uuid update_info.key
 
+let patch_uuid_of_update_uuid uuid =
+  let arr = Uuid.int_array_of_uuid (Uuid.uuid_of_string uuid) in
```

Comment:
I think a lot of my worry is that there might be something we haven't thought of.

For now I suppose the obvious risk to consider is that we might have a collision of patch uuids, though we can avoid this by ensuring that every new update uuid that differs from all previous ones in its first four or last six bytes (since the update uuid can be chosen arbitrarily and need not be random).

Another concern would be collisions of patch uuids with other uuids (for things other than patches) where the same or similar approach has been used. I don't remember for certain whether we have any at present. We might want to add some in future.

What properties are you after here? That to a human reader the uuid-strings should be obviously related (and easy to find one by searching for a chunk of the other) and obviously distinguishable?

Perhaps compromise by just zeroing the `b222` section from the examples above (and ensuring the update uuid has that section non-zero to start with)?

Or we could just swap the `b222` and `c333` sections which I think would make the uuids adequately different to the eye. What's more useful for debugging and log-analysis out of these two properties?
 * being able to perform the forward and reverse transformations manually, vs
 * being able to see at a glance (from the zero chunk) that a uuid is for a patch rather than an update (i.e. to tell which is which, not just tell them apart)

I don't see either of these being obviously better. I think argument for discarding some of the bits relies on the latter property being significantly more valuable, but I'm not convinced that it is.